guide/src/storage/striped_downloads.md

```diff
@@ -156,6 +156,10 @@ Then you read the data and write it to the local file:
 {{#rustdoc_include ../../samples/tests/storage/striped.rs:write-stripe-loop}}
 ```
 
+## Next steps
+
+- Consider optimizing the case where the last stripe only has a few bytes
```

Comment:
comment: if we think in terms of number of stripes, the special case goes away.

We could change the input to be `max_stripe_size`, and then balance the stripes with:

```rs
let stripe_count = metadata.size.div_ceil(max_stripe_size);
let stripe_size = metadata.size / stripe_count;
```

---

I don't know if there are advantages to having "nicely sized stripes", though. (e.g. stripe sizes that are 2^n for some n.)

Reply:
You are right that I didn't handle the remainder. The fix is to round up the stripe size:

```rs
    let stripe_count = size.div_ceil(max_stripe_size);
    let stripe_size = size.div_ceil(stripe_count);
```

> https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=aa6349af995d3acac86181a1d5b66f9f

I disagree that what we want to do is increase the stripe size to 35, when given a max size of 32.

The above snippet yields: `[(0, 26), (26, 26), (52, 26), (78, 26)]`